xoto3/dynamodb/write_versioned/api2.py

```diff
@@ -168,3 +176,20 @@ def create_or_update_trans(vt: VersionedTransaction) -> VersionedTransaction:
         return table.put(creator_updater(table.get(key)(vt)))(vt)
 
     return create_or_update_trans
+
+
+def write_item(
```

Comment:
This is a nice clean implementation, and I like the idea of being able to delegate the decision of whether to update or delete to the writer callable. Does `write_item` clearly communicate that this will delete the record when None is returned?

I might be overthinking this, but the versioned_diff_update code lets you return `None` when no change is desired and that, combined with this being the most general name in the file, might result in someone making some bad assumptions and deleting records... 

It might be worth adding coverage of the `None` behavior in the """ doc (or possibly renaming the function, but "write" is a nice common term for all of the included behavior, so I definitely see the merits...)

Reply:
i've added more verbose documentation around this to the helper - please let me know what else we can add, because I agree that we don't want people accidentally deleting things they don't mean to.

In retrospect, I feel the behavior of versioned_diffed_update_items is one of those premature optimizations for DX - I thought it was 'cute' that returning None could mean "do nothing", but that causes a big problem with regard to composability - you can't transparently compose multiple different updaters if one of them returns `None` to mean "change nothing", because then the next function gets nothing to update. It was an explicit choice to start using None to represent "item does not exist in the table" in this new API, precisely because I wanted to help us make sure we kept everything maximally composable. But documenting this more clearly would be very valuable.